tests/enrollment_token_to_token_test.go

```diff
@@ -0,0 +1,402 @@
+				RegisteredClaims: jwt.RegisteredClaims{},
+				AttributeArray: []string{
+					CustomAttribute1,
+					CustomAttribute2,
+				},
+				AttributeString: CustomAttribute3,
+				CustomId:        CustomId,
+				CustomName:      CustomName,
+			}
+			enrollmentJwt, err := newJwtForExtJwtSigner(extJwtSingerAuthPolicyNoTokens, enrollClaims)
+			ctx.Req.NoError(err)
+			ctx.Req.NotEmpty(enrollmentJwt)
+
+			clientApi := ctx.NewEdgeClientApi(nil)
+			ctx.Req.NotNil(clientApi)
+
+			err = clientApi.CompleteJwtTokenEnrollmentToTokenAuth(enrollmentJwt)
+
+			ctx.Req.Error(err)
+		})
+
+		t.Run("if the token is missing", func(t *testing.T) {
+			ctx.testContextChanged(t)
+			clientApi := ctx.NewEdgeClientApi(nil)
+			ctx.Req.NotNil(clientApi)
+
+			err := clientApi.CompleteJwtTokenEnrollmentToTokenAuth("")
+
+			ctx.Req.Error(err)
+		})
+
+		t.Run("if the token is malformed", func(t *testing.T) {
+			ctx.testContextChanged(t)
+			clientApi := ctx.NewEdgeClientApi(nil)
+			ctx.Req.NotNil(clientApi)
+
+			err := clientApi.CompleteJwtTokenEnrollmentToTokenAuth("eYsdfgsdgfd.dfgsfdgfds.sdfgsdfg")
+
+			ctx.Req.Error(err)
+		})
+
+		t.Run("if the token is from another signer", func(t *testing.T) {
+			ctx.testContextChanged(t)
+
+			extJwtSingerEnrollToTokenValid := createExtJwtComponents("enroll-to-token-valid-signer")
+			extJwtSingerEnrollToTokenValid.Create.EnrollToTokenEnabled = true
+			extJwtSingerEnrollToTokenValid.Create.EnrollAuthPolicyID = *authPolicyOnlyExtJwtCreate.Detail.ID
+			extJwtSingerEnrollToTokenValid.Create.EnrollAttributeClaimsSelector = ""
+			extJwtSingerEnrollToTokenValid.Create.ClaimsProperty = nil
+			extJwtSingerEnrollToTokenValid.Create.EnrollNameClaimsSelector = ""
+			extJwtSingerEnrollToTokenValid.Detail, err = adminManClient.CreateExtJwtSigner(extJwtSingerEnrollToTokenValid.Create)
+			ctx.Req.NoError(err)
+			ctx.Req.NotNil(extJwtSingerEnrollToTokenValid.Detail)
+
+			extJwtSingerEnrollToTokenWrongSigner := createExtJwtComponents("enroll-to-token-invalid-wrong-singer")
+			extJwtSingerEnrollToTokenWrongSigner.Create.EnrollAuthPolicyID = *authPolicyOnlyExtJwtCreate.Detail.ID
+			extJwtSingerEnrollToTokenWrongSigner.Create.EnrollAttributeClaimsSelector = ""
+			extJwtSingerEnrollToTokenWrongSigner.Create.ClaimsProperty = nil
+			extJwtSingerEnrollToTokenWrongSigner.Create.EnrollNameClaimsSelector = ""
+			extJwtSingerEnrollToTokenWrongSigner.Detail, err = adminManClient.CreateExtJwtSigner(extJwtSingerEnrollToTokenWrongSigner.Create)
+			ctx.Req.NoError(err)
+			ctx.Req.NotNil(extJwtSingerEnrollToTokenWrongSigner.Detail)
+
+			enrollClaims := &claimsWithAttributes{}
+			enrollmentJwt, err := newJwtForExtJwtSigner(extJwtSingerEnrollToTokenWrongSigner, enrollClaims)
+			ctx.Req.NoError(err)
+			ctx.Req.NotEmpty(enrollmentJwt)
+
+			clientApi := ctx.NewEdgeClientApi(nil)
+			ctx.Req.NotNil(clientApi)
+
+			err = clientApi.CompleteJwtTokenEnrollmentToTokenAuth(enrollmentJwt)
+
+			ctx.Req.Error(err)
+		})
+	})
```

Comment:
i see a lot of Req.NoError tests around EnrollNameClaimsSelector but are there any negative tests? What if i pass `/` or a field that doesn't exist or a number when it should be a string etc?